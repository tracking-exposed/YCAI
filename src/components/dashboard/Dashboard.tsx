import { Typography, useTheme } from '@material-ui/core';
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';
import * as QR from 'avenger/lib/QueryResult';
import { declareQueries } from 'avenger/lib/react';
import { pipe } from 'fp-ts/lib/function';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { auth, localProfile } from '../../state/creator.queries';
import { CurrentView, currentView } from '../../utils/location.utils';
import { ErrorBox } from '../common/ErrorBox';
import { LazyFullSizeLoader } from '../common/FullSizeLoader';
import Settings from './Settings';
import { StatisticsPage } from './community/StatisticsPage';
import { LinkAccount } from './LinkAccount';
import { Sidebar } from './Sidebar';
import { Lab } from './lab/Lab';
import { LabVideoEdit } from './lab/LabVideoEdit';
import { ContentCreator } from '@backend/models/ContentCreator';
import { AuthResponse } from '@backend/models/Auth';

const useStyles = makeStyles((theme) => ({
  root: {
    height: '100%',
    width: '100%',
    backgroundColor: theme.palette.background.default,
  },
  title: {
    marginBottom: 15,
  },
  subtitle: {
    marginBottom: 20,
  },
}));

interface DashboardContentProps {
  currentView: CurrentView;
  profile: ContentCreator | null;
  auth: AuthResponse | null;
}

const DashboardContent: React.FC<DashboardContentProps> = ({
  currentView,
  profile,
  auth,
}) => {
  const { t } = useTranslation();
  const theme = useTheme();
  const classes = useStyles();

  const [currentViewLabel, currentViewSubtitle, currentViewContent] =
    React.useMemo(() => {
      switch (currentView.view) {
        case 'settings':
          // eslint-disable-next-line react/jsx-key
          return [t('routes:settings'), '', <Settings />];
        default: {
          if (profile === null) {
            return [
              t('routes:link_account'),
              t('link_account:subtitle'),
              // eslint-disable-next-line react/jsx-key
              <LinkAccount auth={auth} />,
            ];
          }

          switch (currentView.view) {
            case 'labEdit':
              return [
                t('routes:lab_title'),
                t('routes:lab_edit_subtitle'),
                // eslint-disable-next-line react/jsx-key
                <LabVideoEdit videoId={currentView.videoId} />,
              ];
            case 'lab':
              return [
                t('routes:lab_title'),
                t('routes:lab_subtitle'),
                // eslint-disable-next-line react/jsx-key
                <Lab />,
              ];
            default:
              return [
                t('routes:statistics'),
                t('statistics:subtitle'),
                // eslint-disable-next-line react/jsx-key
                <StatisticsPage />,
              ];
          }
        }
      }
    }, [currentView, auth]);

  return (
    <Grid
      container
      alignContent="flex-start"
      style={{
        padding: theme.spacing(2),
        overflowY: 'scroll',
        height: '100%',
        width: '100%',
      }}
    >
      <Grid item>
        <Typography variant="h1" color="primary" className={classes.title}>
          {currentViewLabel}
        </Typography>
        <Typography
          variant="subtitle1"
          color="textPrimary"
          className={classes.subtitle}
        >
          {currentViewSubtitle}
        </Typography>
      </Grid>
      <Grid item>{currentViewContent}</Grid>
    </Grid>
  );
};

const withQueries = declareQueries({
  currentView,
  profile: localProfile,
  auth,
});

export const Dashboard = withQueries(({ queries }): React.ReactElement => {
  return pipe(
    queries,
    QR.fold(LazyFullSizeLoader, ErrorBox, ({ currentView, profile, auth }) => {
      const theme = useTheme();
      const classes = useStyles();

      return (
        <Grid container className={classes.root}>
          <Grid
            item
            lg={2}
            md={3}
            sm={3}
            style={{ padding: theme.spacing(2), height: '100%' }}
          >
            <Sidebar currentView={currentView} />
          </Grid>
          <Grid item lg={10} md={9} sm={9} style={{ height: '100%' }}>
            <DashboardContent
              currentView={currentView}
              profile={profile}
              auth={auth}
            />
          </Grid>
        </Grid>
      );
    })
  );
});
