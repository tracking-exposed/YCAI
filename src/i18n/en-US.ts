import { CustomTypeOptions } from 'react-i18next';

const resources: CustomTypeOptions['resources'] = {
  title: 'YouChoose AI',
  common: {
    coming_soon: 'Coming soon',
    empty_list: 'No {{resource}} found.',
  },
  actions: {
    popup_bootstrap: 'Bootstrap',
    add: 'Add',
    manage_recommendations: 'Manage recommendations',
    link_channel: 'Link channel',
    unlink_channel: 'Unlink channel',
    delete: 'Delete',
    clear: 'Clear',
    editThisVideo: 'Edit this video',
    importVideos: 'Import videos',
    addToCurrentVideo: 'Add to current video',
    removeFromCurrentVideo: 'Remove from current video',
    compare: 'Compare',
    related: 'Related',
    verify_channel: 'Verify channel',
    generate_keypair: 'Generate keypair',
    refresh_keypair: 'Refresh keypair',
    delete_keypair: 'Delete keypair',
    download_keypair: 'Download keypair',
    pull_creator_videos: 'Import your videos',
    update_creator_videos_list: 'Update your list of videos',
    copy_verification_code: 'Copy code',
    unlink_profile: 'Unlink profile',
    edit_access_token: 'Edit Access Token',
    download_access_token: 'Download Access Token',
    download: 'Download'
  },
  routes: {
    studio: 'Studio',
    statistics: 'Statistics',
    settings: 'Settings',
    link_account: 'Link your account',
  },
  account: {
    channel: 'Your channel',
  },
  creator: {
    title: 'Creator',
  },
  link_account: {
    title: 'Link your account',
    subtitle: 'Our system will verify that you own the channel on this browser',
    label: 'Link your account to start choosing recommendations',
    verification_code_label: 'Verification code',
    verification_code_hint:
      "Copy the code below and add it to your YouTube channel's description",
  },
  youtube: {
    title: 'Youtube',
  },
  recommendations: {
    added_to_video_title: 'Recommendations added',
    by_creator_title: 'Author\'s recommendations',
    total: 'Total recommendations',
    url: 'Recommendation url',
    yours: 'Your recommendations',
    no_items: 'No recommendations found',
    add_to_video: 'Add a recommendation to this video',
    url_placeholder: 'https://youtube.com/watch?v=xxxxx',
    url_helper_text: 'Insert a link',
  },
  dashboard: { title: 'Dashboard' },
  popup: { version: 'version {{version}} build {{data}}' },
  statistics: {
    title: 'Statistics',
    subtitle: 'Statistics computed with resources from other users',
    recommendability_score_title: 'Recommendability Score',
    recommendability_score_subtitle:
      'Where your videos appears as recommended?',
    total_views: 'Total Views',
    total_recommendations: 'Total Recommendations',
    evidences_title: 'Evidences',
    notifications_title: 'Notifications',
    top_n_cc_related_to_your_channel:
      'Top {{count}} CC related to your channel',
    advertising_connected_to_your_videos:
      'Advertising connected to your videos',
  },
  videos: {
    no_results: 'No videos found.',
    no_selected: 'No video selected',
    no_video_id: 'No video id found',
  },
  settings: {
    contentCreatorRecommendationLabel: 'Content Creators',
    contentCreatorRecommendationHint: 'See suggestions by real authors',
    communityRecommendationsLabel: 'Community',
    communityRecommendationsHint: 'Coming soon ðŸŒ»',
    contributeToIndipendentStatsLabel: 'Independent stats',
    contributeToIndipendentStatsHint:
      'Donate anonymously what Youtube recommends and advertises you',
    api_list_title: 'API List',
    encrypted_contributions_private_key: 'You can download your private key in dashboard.',
    keypair_passphrase: 'Key pair passphrase',
    keypair_private_key: 'Key pair secret key',
    keypair_public_key: 'Key pair public key',
    keypair_title: 'Keypair for independent contribution',
    access_token_title: 'You Access Token',
    access_token: 'Access Token'
  },
  ytVideoPage: {
    firstTab: 'Creator Recommendations',
    secondTab: 'Community Recommendations',
    thirdTab: 'Youtube Recommendations',
  },
};

export default resources;
