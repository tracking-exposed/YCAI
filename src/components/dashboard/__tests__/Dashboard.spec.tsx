import React from 'react';
import renderer from 'react-test-renderer';
import { Dashboard } from '../Dashboard';
import '../../../i18n';

describe('Dashboard', () => {
  test('Should mount the dashboard', () => {
    const component = renderer.create(<Dashboard />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
