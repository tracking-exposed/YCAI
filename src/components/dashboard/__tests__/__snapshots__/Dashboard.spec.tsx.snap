// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Should mount the dashboard 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "100%",
      "justifyContent": "center",
      "minHeight": 600,
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <span
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </span>
</div>
`;
